codex-rs/mcp-server/src/wire_format.rs

```diff
@@ -286,4 +325,72 @@ mod tests {
             serde_json::to_value(&request).unwrap(),
         );
     }
+
+    #[test]
```

Comment:
FYI, I added a couple of tests originally just to verify I put `#[serde(tag = "method", rename_all = "camelCase")]` in the right place, but I don't know that we have to verify the serialized version of every message type. I think I have a better way to verify this sort of thing going forward, so don't feel like you have to add tests here to match the above, but feel free to keep them if you find them helpful.